src/plugins/kibana/server/routes/api/index_pattern_endpoints.js

```diff
@@ -0,0 +1,258 @@
+const esErrors = require('elasticsearch').errors;
+const Boom = require('boom');
+const Joi = require('joi');
+const _ = require('lodash');
+const Promise = require('bluebird');
+const getMappings = require('../../lib/get_mappings');
+const stitchPatternAndMappings = require('../../lib/stitch_pattern_and_mappings');
+const {templateToPattern, patternToTemplate} = require('../../lib/convert_pattern_and_template_name');
+const removeDeprecatedFieldProps = require('../../lib/remove_deprecated_field_props');
+const indexPatternSchema = require('../../lib/schemas/index_pattern_schema');
+
+export default function (server) {
+
+  let handleESError = function (error) {
```

Comment:
Also this function seems useful for a lot more than the index pattern API, probably belongs in `kibana/server/lib`